### PR TITLE
fixed CaloTaus (memory issue of ECALBounds) for CMSSW 7_4_x

### PR DESCRIPTION
**Pull request by veelken**
bug-fix for problem reported in this Hypernews thread:
https://hypernews.cern.ch/HyperNews/CMS/get/swDevelopment/3218

### CODE REVIEW

File: RecoTauTag/TauTagTools/interface/ECALBounds.h [hunk 1]
```diff
@@ -1,41 +1,48 @@
-#ifndef RecoTauTag_TauTagTools_ECALBounds_H
-#define RecoTauTag_TauTagTools_ECALBounds_H
+#ifndef GeomPropagators_ECALBounds_H
```
Comment (slava77) [inline RecoTauTag/TauTagTools/interface/ECALBounds.h]: please keep the include guards match the full name of the file: it's not a GeomPropagators package here

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@cmsbuild please test
the issues with tests with PU were resolved

**Comment by cvuosalo** [comment]
+1

For #9477 79f6048fbd6109d70d087473942cdb4202e21aaa

Revised CaloTau reco sequence to fix memory bug that showed up in 74X. This non-standard sequence seems to be mostly obsolete and is known to be in use by only one analysis (for Run 1), so the Tau POG may decide to eliminate it entirely in the near future. This PR should not change standard workflows.

The code changes are satisfactory, and Jenkins [tests](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9477/5351/summary.html) against baseline CMSSW_7_5_X_2015-06-06-2300 show no significant differences, as expected.

**Comment by davidlange6** [comment]
+1

**Pull request merged**